contentcuration/contentcuration/tests/viewsets/test_contentnode.py

```diff
@@ -1011,3 +1006,137 @@ def test_delete_orphanage_root(self):
+    def test_descendant_count(self):
+        topic_tree_node = testdata.tree()
+        self.channel.main_tree = topic_tree_node
+        self.channel.save()
+        response = self.client.get(
+            reverse("contentnode-detail", kwargs={"pk": topic_tree_node.id})
+        )
+        serialized = response.data
+
+        self.assertEqual(serialized.get("total_count"), 7)
+
+    def test_assessment_count(self):
+        node = models.ContentNode.objects.get(
+            node_id="00000000000000000000000000000005"
+        )
+        self.channel.main_tree = node
+        self.channel.save()
+        response = self.client.get(
+            reverse("contentnode-detail", kwargs={"pk": node.id})
+        )
+        serialized = response.data
+
+        self.assertEqual(serialized.get("assessment_item_count"), 3)
+
+    def test_resource_count(self):
+        topic_tree_node = testdata.tree()
+        self.channel.main_tree = topic_tree_node
+        self.channel.save()
+        response = self.client.get(
+            reverse("contentnode-detail", kwargs={"pk": topic_tree_node.id})
+        )
+        serialized = response.data
+
+        self.assertEqual(serialized.get("resource_count"), 5)
+
+    def test_coach_count(self):
+        topic_tree_node = testdata.tree()
+        self.channel.main_tree = topic_tree_node
+        self.channel.save()
+        nested_topic = (
+            topic_tree_node.get_descendants().filter(kind=content_kinds.TOPIC).first()
+        )
+        self.create_coach_node(nested_topic)
+
+        response = self.client.get(
+            reverse("contentnode-detail", kwargs={"pk": topic_tree_node.id})
+        )
+        serialized = response.data
+
+        self.assertEqual(1, serialized.get("coach_count"))
+
+    def test_multiple(self):
+        topic_tree_node1 = testdata.tree()
+        topic_tree_node2 = testdata.tree()
+
+        topic_tree1_topics = topic_tree_node1.get_descendants().filter(
+            kind=content_kinds.TOPIC
+        )
+        video_node = self.create_coach_node(topic_tree1_topics.first())
+        exercise_node = models.ContentNode.objects.get(
+            tree_id=topic_tree_node2.tree_id, node_id="00000000000000000000000000000005"
+        )
+
+        self.set_tree_changed(topic_tree_node1, False)
+        self.set_tree_changed(topic_tree_node2, False)
+        topic_tree1_topics.last().delete()
+
+        topic_tree1_results = self.fetch_data(topic_tree_node1)
+        topic_tree2_results = self.fetch_data(topic_tree_node2)
+        video_node_results = self.fetch_data(video_node)
+        exercise_node_results = self.fetch_data(exercise_node)
+
+        self.assertIsNotNone(topic_tree1_results)
+        self.assertEqual(6, topic_tree1_results.get("total_count"))
+        self.assertEqual(5, topic_tree1_results.get("resource_count"))
+        self.assertEqual(0, topic_tree1_results.get("assessment_item_count"))
+        self.assertEqual(1, topic_tree1_results.get("coach_count"))
```

Comment:
I also recently added new_count and updated_count

Reply:
I was preserving these while doing some cleanup, but we can add more tests in future!